internal/convert/registryv1.go

```diff
@@ -189,10 +185,7 @@ func Convert(in RegistryV1, installNamespace string, targetNamespaces []string)
 			supportedInstallModes.Insert(string(im.Type))
 		}
 	}
-	if !supportedInstallModes.Has(string(v1alpha1.InstallModeTypeAllNamespaces)) {
-		return nil, fmt.Errorf("AllNamespace install mode must be enabled")
-	}
-	if targetNamespaces == nil {
+	if len(targetNamespaces) == 0 {
 		if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeAllNamespaces)) {
 			targetNamespaces = []string{""}
 		} else if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeOwnNamespace)) {
```

Comment:
Should we start requiring that for `OwnNamespace`, the targetNamespace must be explicitly specified?

Reply:
I'm fine with it, will it break any existing norm that v0 has set in terms of if `OwnNamespace` is set, we watch installNs.

Reply:
In v0, the OperatorGroup explicitly lists (or matches) just the namespace that it is in, so it is explicit in that sense.

The reason this logic exists in rukpak is because (until now) there is no way to define a list of target namespaces, so they only possibilities we could even derive were AllNamespaces or OwnNamespace.

Reply:
Leaving it as is for now based on the [slack](https://kubernetes.slack.com/archives/C0181L6JYQ2/p1705688983270469?thread_ts=1705669480.907989&cid=C0181L6JYQ2) discussion.